Crash when cancelling podcast search

```
io.reactivex.exceptions.UndeliverableException: 
  at io.reactivex.plugins.RxJavaPlugins.onError (RxJavaPlugins.java:367)
  at io.reactivex.internal.operators.single.SingleCreate$Emitter.onError (SingleCreate.java:81)
  at io.reactivex.internal.operators.single.SingleCreate.subscribeActual (SingleCreate.java:42)
  at io.reactivex.Single.subscribe (Single.java:3438)
  at io.reactivex.internal.operators.single.SingleDoOnDispose.subscribeActual (SingleDoOnDispose.java:38)
  at io.reactivex.Single.subscribe (Single.java:3438)
  at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run (SingleSubscribeOn.java:89)
  at io.reactivex.Scheduler$DisposeTask.run (Scheduler.java:578)
  at io.reactivex.internal.schedulers.ScheduledRunnable.run (ScheduledRunnable.java:66)
  at io.reactivex.internal.schedulers.ScheduledRunnable.call (ScheduledRunnable.java:57)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201 (ScheduledThreadPoolExecutor.java:154)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:269)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)
Caused by: java.lang.InterruptedException: 
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.doAcquireSharedInterruptibly (AbstractQueuedSynchronizer.java:972)
  at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly (AbstractQueuedSynchronizer.java:1278)
  at java.util.concurrent.CountDownLatch.await (CountDownLatch.java:203)
  at de.danoeh.antennapod.discovery.CombinedSearcher.lambda$search$2 (CombinedSearcher.java:53)
  at de.danoeh.antennapod.discovery.-$$Lambda$CombinedSearcher$kqeC9TFOHWl-icT90ge38FWa8vM.subscribe (lambda)
  at io.reactivex.internal.operators.single.SingleCreate.subscribeActual (SingleCreate.java:39)
  at io.reactivex.Single.subscribe (Single.java:3438)
  at io.reactivex.internal.operators.single.SingleDoOnDispose.subscribeActual (SingleDoOnDispose.java:38)
  at io.reactivex.Single.subscribe (Single.java:3438)
  at io.reactivex.internal.operators.single.SingleSubscribeOn$SubscribeOnObserver.run (SingleSubscribeOn.java:89)
  at io.reactivex.Scheduler$DisposeTask.run (Scheduler.java:578)
  at io.reactivex.internal.schedulers.ScheduledRunnable.run (ScheduledRunnable.java:66)
  at io.reactivex.internal.schedulers.ScheduledRunnable.call (ScheduledRunnable.java:57)
  at java.util.concurrent.FutureTask.run (FutureTask.java:237)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201 (ScheduledThreadPoolExecutor.java:154)
  at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run (ScheduledThreadPoolExecutor.java:269)
  at java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1113)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:588)
  at java.lang.Thread.run (Thread.java:818)
```

--- a/app/src/main/java/de/danoeh/antennapod/discovery/ItunesTopListLoader.java
+++ b/app/src/main/java/de/danoeh/antennapod/discovery/ItunesTopListLoader.java
@@ -33,16 +33,12 @@ public class ItunesTopListLoader {
             OkHttpClient client = AntennapodHttpClient.getHttpClient();
             String feedString;
             try {
-                try {
-                    feedString = getTopListFeed(client, lang, limit);
-                } catch (IOException e) {
-                    feedString = getTopListFeed(client, "us", limit);
-                }
-                List<PodcastSearchResult> podcasts = parseFeed(feedString);
-                emitter.onSuccess(podcasts);
-            } catch (IOException | JSONException e) {
-                emitter.onError(e);
+                feedString = getTopListFeed(client, lang, limit);
+            } catch (IOException e) {
+                feedString = getTopListFeed(client, "us", limit);
             }
+            List<PodcastSearchResult> podcasts = parseFeed(feedString);
+            emitter.onSuccess(podcasts);
         })
                 .subscribeOn(Schedulers.io())
                 .observeOn(AndroidSchedulers.mainThread());

--- a/core/src/free/java/de/danoeh/antennapod/core/ClientConfig.java
+++ b/core/src/free/java/de/danoeh/antennapod/core/ClientConfig.java
@@ -34,7 +34,7 @@ public class ClientConfig {
     private static boolean initialized = false;
 
     public static synchronized void initialize(Context context) {
-        if(initialized) {
+        if (initialized) {
             return;
         }
         PodDBAdapter.init(context);
@@ -42,6 +42,7 @@ public class ClientConfig {
         PlaybackPreferences.init(context);
         NetworkUtils.init(context);
         SleepTimerPreferences.init(context);
+        RxJavaErrorHandlerSetup.setupRxJavaErrorHandler();
         initialized = true;
     }
 

--- a/core/src/main/java/de/danoeh/antennapod/core/util/exception/RxJavaErrorHandlerSetup.java
+++ b/core/src/main/java/de/danoeh/antennapod/core/util/exception/RxJavaErrorHandlerSetup.java
@@ -1,0 +1,26 @@
+package de.danoeh.antennapod.core.util.exception;
+
+import android.util.Log;
+import io.reactivex.exceptions.UndeliverableException;
+import io.reactivex.plugins.RxJavaPlugins;
+
+public class RxJavaErrorHandlerSetup {
+
+    private RxJavaErrorHandlerSetup() {
+
+    }
+
+    public static void setupRxJavaErrorHandler() {
+        RxJavaPlugins.setErrorHandler(e -> {
+            if (e instanceof UndeliverableException) {
+                e = e.getCause();
+            }
+            if (e instanceof InterruptedException) {
+                // fine, some blocking code was interrupted by a dispose call
+                Log.d("RxJavaErrorHandler", "Ignored exception: " + Log.getStackTraceString(e));
+                return;
+            }
+            Thread.currentThread().getUncaughtExceptionHandler().uncaughtException(Thread.currentThread(), e);
+        });
+    }
+}

--- a/core/src/play/java/de/danoeh/antennapod/core/ClientConfig.java
+++ b/core/src/play/java/de/danoeh/antennapod/core/ClientConfig.java
@@ -8,6 +8,7 @@ import de.danoeh.antennapod.core.preferences.SleepTimerPreferences;
 import de.danoeh.antennapod.core.preferences.UserPreferences;
 import de.danoeh.antennapod.core.storage.PodDBAdapter;
 import de.danoeh.antennapod.core.util.NetworkUtils;
+import de.danoeh.antennapod.core.util.exception.RxJavaErrorHandlerSetup;
 
 /**
  * Stores callbacks for core classes like Services, DB classes etc. and other configuration variables.
@@ -36,7 +37,7 @@ public class ClientConfig {
     private static boolean initialized = false;
 
     public static synchronized void initialize(Context context) {
-        if(initialized) {
+        if (initialized) {
             return;
         }
         PodDBAdapter.init(context);
@@ -45,6 +46,7 @@ public class ClientConfig {
         NetworkUtils.init(context);
         CastManager.init(context);
         SleepTimerPreferences.init(context);
+        RxJavaErrorHandlerSetup.setupRxJavaErrorHandler();
         initialized = true;
     }
 

